### ISSUE TITLE
Could not process file due to BetterReflection\ReflectionClass "Carbon\Traits\Localization" could not be found in the located source (with POC)

### ISSUE THREAD

**Issue by ruudk** [rectorphp/rector#4406]
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | Rector 0.8.23              |
| Installed as   | rector/rector-prefixed                      |

Somehow Rector trips when using [Carbon](https://github.com/briannesbitt/Carbon) together with `AUTO_IMPORT_NAMES`.

```
$ vendor/bin/rector process src --dry-run
Rector 0.8.23
Config file: rector.php

 4/4 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

1 file with changes
===================

1) src/Entity.php

    ---------- begin diff ----------
--- Original
+++ New
@@ -4,13 +4,11 @@

 namespace MyNamespace;

-use Carbon\Carbon;
-
+use Carbon;
 class Entity
 {
     private Carbon $createdAt;
-
-    public function __construct()
+    public function __construct()
     {
         $this->createdAt = Carbon::now();
     }
    ----------- end diff -----------



 [ERROR] Could not process "src/Entity.php" file, due to:
         "_HumbugBox2e5f015a1738\Roave\BetterReflection\Reflection\ReflectionClass "Carbon\Traits\Localization" could
         not be found in the located source".

```

## Minimal PHP Code Causing Issue

Please see working POC repository here: https://github.com/ruudk/rector-carbon-bug

## Expected Behaviour

Not crash.

**Comment by ruudk** [comment]
I just found an issue posted on the BetterReflection repo https://github.com/Roave/BetterReflection/issues/689

**Comment by TomasVotruba** [comment]
Hi,

could you share a link to this `Localization` trait on Github?

**Comment by ruudk** [comment]
It's located here https://github.com/briannesbitt/Carbon/blob/c762e8994a0467b24fa17dd8f10bc52d375e75ff/src/Carbon/Traits/Localization.php#L24-L29

**Comment by TomasVotruba** [comment]
Ah yes, that's what I remember. So we basically patch that part to be removed.

**Comment by ruudk** [comment]
Then I think it would be better to submit a PR to Carbon to get that stuff removed.

**Comment by TomasVotruba** [comment]
Also  a way. Depends how fast you need it

**Comment by ERuban** [comment]
Hi,

have stacked with this issue too in our projects and have fixed it in a bit another way. 

So, my steps to fix it was:
- downgrade and fix version in composer.json for "nikic/php-parser": "4.9.*"
- downgrade and fix version in composer.json for "phpstan/phpstan": "0.12.42"
- downgrade and fix version in composer.json for "rector/rector": "0.8.7" - the latest version that uses nikic/php-parser:4.9.0

**Comment by TomasVotruba** [comment]
This is the best solution so far, untill it gets fixed in carbon
https://github.com/rectorphp/rector/issues/4406#issuecomment-708506993

**Comment by u01jmg3** [comment]
Is Carbon willing to apply a fix? I don't see an open issue on their side.

**Comment by ruudk** [comment]
I didn't open an issue because I saw they were working on a new major that will prob fix this. But I'm not sure. Maybe good to open the issue anyway?

**Comment by u01jmg3** [comment]
Might be a good idea to get it on their radar and see if it is something they are willing to even fix.

**Comment by ruudk** [comment]
@u01jmg3 Go ahead :)

**Comment by BoGnY** [comment]
Same problem here.. All Laravel project have this problem

**Comment by HDVinnie** [comment]
This appears to be fixed on `Rector dev-master@8a64821`

**Comment by samsonasik** [comment]
@HDVinnie thank you for verify. I am closing it.
